hail/python/test/hailtop/batch/test_batch.py

```diff
@@ -521,3 +540,18 @@ def test_envvar(self):
         j.env('SOME_VARIABLE', '123abcdef')
         j.command('[ $SOME_VARIABLE = "123abcdef" ]')
         assert b.run().status()['state'] == 'success'
+
+    def test_single_job_with_shell(self):
+        msg = 'hello world'
+        b = self.batch()
+        j = b.new_job(shell='/bin/sh')
+        j.command(f'echo "{msg}"')
+        r = b.run(wait=True)
+        res = self.backend._batch_client.get_job_log(r.id, b._jobs[0]._job_id)["main"].rstrip()
```

Comment:
@jigold do we intend to let Batch users query the status and log of jobs?  I don't see it in the interface and obviously the code Alex had to write here is far from ideal.

Reply:
OK, to keep this moving forward, while we're working on the log story with Jackie, I think we should just change this to match the other batch tests and check `assert b.run().status()['state'] == 'success'` and follow up with a more precise test when that's sorted.

Reply:
Thanks for bringing this up! I think I can expose it on a job and batch object, but there isn't a good way to query the status and logs for a batch in a Python script where the object no longer exists. For example, doing queries after the fact a week later. I can put it on Asana to think this through more carefully. I think it would involve adding functionality to the ServiceBackend that takes either a Job or Batch object or the batch_id, job_id and returns the logs and status etc rather than having it as attributes on the batch or Job object directly.

Reply:
> object no longer exists

You mean locally?  Yeah, I thought that might be why this wasn't there already.  Some options:

 - The LocalBackend can't support it, so do nothing.
 - Don't delete the docker images in the local backend (e.g. `docker run` without `--rm`) so they can be queried afterwards.  Maybe make it an option in LocalBackend.  I'm not sure how cleanup gets handled.  Maybe have a hailctl tool to query the docker containers (do they have labels?) to see which batches are there and let you delete them.
 - Capture the docker state in the local backend before deleting.  Doing this with the logs seems problematic.
 - Don't change the interface, but make private methods to get status and log on the service backend that we can use in the tests.
 - Have the log and status functions throw an support exception in the local backend.

Has anyone asked for a local batch feature like this?  An internal functions for the service backend seem like a no brainer.  Then we can think about how/whether we should support them on the local backend.

Separately, how do you know if a LocalBackend batch suceeds or fails?  It looks like LocalBackend._run doesn't return.  We should probably document more clearly what the Batch.run return value is.